synapse/handlers/room.py

```diff
@@ -683,13 +683,20 @@ async def create_room(
 
         directory_handler = self.hs.get_handlers().directory_handler
         if room_alias:
-            await directory_handler.create_association(
-                requester=requester,
-                room_id=room_id,
-                room_alias=room_alias,
-                servers=[self.hs.hostname],
-                check_membership=False,
+            # Check if publishing is blocked by a third party module
+            allowed_by_third_party_rules = await (
+                self.third_party_event_rules.check_room_can_be_added_to_public_rooms_directory(
+                    room_id
+                )
             )
+            if allowed_by_third_party_rules:
+                await directory_handler.create_association(
+                    requester=requester,
+                    room_id=room_id,
+                    room_alias=room_alias,
+                    servers=[self.hs.hostname],
+                    check_membership=False,
+                )
```

Comment:
adding things to the room directory, and setting an alias, are different things?

Reply:
Oops, yep this was a mistake stemming from the `.../directory/list/room/...` and `.../directory/room/...` APIs looking similar...

Thanks for the catching that! Seems like I need to add an extra test case as well.